### PR TITLE
feat(tests): #930 api test coverage

### LINKED ISSUE #930: Add code coverage to API tests
Need to get coverage working against serverless....

> Tip: Use [nyc](https://github.com/istanbuljs/nyc) to do this....

### PR DESCRIPTION
**Pull request by eddiejaoude**
closes #930 

### Notes

A summary of what was achieved in this PR

- [ ] API: Task One (eg. updated `deployed` model )
- [ ] UI: Task Three (eg. added validation rules on create environment page)
- [ ] Documentation: Updated accordingly

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -17,6 +17,15 @@
   "bugs": {
     "url": "https://github.com/DashboardHub/PipelineDashboard/issues"
   },
+  "scripts": {
+    "install:local": "npx serverless dynamodb install --stage dev",
```
Comment (andrew-cunliffe) [inline api/package.json]: don't need any `npx`in package.json - all commands here assume usage of the installed modules automatically

Comment (andrew-cunliffe) [inline api/package.json]: don't need any npxin package.json - all commands here assume usage of the installed modules automatically

File: README.md [hunk 2]
```diff
@@ -82,13 +82,14 @@ Then visit `http://localhost:4200/`
 
 This is done using `CumcumberJS`, with the `Given`, `When`, `Then` format in English, in the `api/tests/features` directory.
 
-Internet is needed when running the automated API tests, because a real `jwt` is used from `Auth0` service.
+**Note**
+
+- Internet is needed when running the automated API tests, because a real `jwt` is used from `Auth0` service.
+- The API must NOT be running
```
Comment (andrew-cunliffe) [inline README.md]: could consider running API on a different port to allow parallel development and testing

File: api/package.json (hunk 1 above)
Comment (eddiejaoude) [reply api/package.json]: Ok great 👍

File: README.md (hunk 2 above)
Comment (eddiejaoude) [reply README.md]: Good idea, probably for a future improvement 🤓

**Comment by eddiejaoude** [comment]
Not sure why CI is not happy, I will investigate. Thanks for reviewing.

File: Makefile [hunk 3]
```diff
@@ -56,42 +56,43 @@ api.env.test: guard-AUTH0_CLIENT_ID_TEST guard-AUTH0_CLIENT_SECRET_TEST api.clea
 	(cd api; sed -i 's|pipelinedashboard-projects|pipelinedashboard-projects-test|g' ./config.json)
 
 api.test:
-	(cd api; ./node_modules/.bin/cucumber-js ./tests/features --require ./tests --format ./node_modules/cucumber-pretty --tags ${TEST_TAGS})
+	(cd api; rm -fr coverage)
+	(cd api; npm test)
```
Comment (andrew-cunliffe) [inline Makefile]: should be `npm run test` this is why CI fails I think....

Comment (eddiejaoude) [reply Makefile]: I thought `test` was a special one? It works locally, but I can change it just in case. Thanks for the tip 🤓

Comment (andrew-cunliffe) [reply Makefile]: I did think that as I wrote, I am just used to using run for everything except start :)

Comment (eddiejaoude) [reply Makefile]: yeah probably better to have it in front of most things as its less likely to cause an issue in the future

**Pull request closed without merging**